CONTRIBUTING.md

```diff
@@ -0,0 +1,324 @@
+| _model_name/tokenization.py_ (optional) | Script containing the model tokenizer class which inherits from HuggingFace's [`PretrainedTokenizer`](https://huggingface.co/transformers/main_classes/tokenizer.html#transformers.PreTrainedTokenizer) class or its family of derived classes. |
+| _model_name/train.py_                   | Script containing code to train the model. It is recommended to utilize the [`Trainer`](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer) class from HuggingFace.                                             |
+| _model_name/README.md_                  | README markdown file containing model information such as model source, architecture, evaluation datasets and metrics, model size, and training information.                                                                                    |
+| _model_name/requirements.txt_           | Text file containing package dependencies for model serving. To manage the full list of dependencies in `sgnlp`, contributors are strongly recommended to limit their code to use the packages listed in `setup.py`.                            |
+
+### Config
+
+Model configs contain model architecture information. Typically, this would include hyperparameters for the different layers within the model as well as the loss function. Model configs should inherit from the `PretrainedConfig` class from the `transformers` package. The following is an example from the Cross Lingual Cross Domain Sentiment Analysis model.
+
+```
+from transformers import PretrainedConfig
+
+class UFDClassifierConfig(PretrainedConfig):
+
+    model_type = "classifier"
+
+    def __init__(self, embed_dim=1024, num_class=2, initrange=0.1, **kwargs):
+        super().__init__(**kwargs)
+        self.embed_dim = embed_dim
+        self.num_class = num_class
+        self.initrange = initrange
+```
+
+For models that use or adapt pre-trained configs already available in the `transformers` package, the model config should inherit from the pre-trained config class instead. For example, this model config inherits from `BertConfig` which is a child class of `PretrainedConfig`.
+
+```
+from transformers import BertConfig
+
+class NewModelConfig(BertConfig):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+```
+
+### Preprocess
+
+The `preprocess.py` script and its associated preprocessor class is an addition in `sgnlp`. When implementing various models, the team found that more complex models required more complex preprocessing. For example, some NLP models take in multiple different text inputs (ie, different utterances, multiple tweets, a single question and multiple possible answers, etc) which require different preprocessing steps. The `preprocess.py` and the preprocessor class is the team's solution to packaging all of these different steps into a single and consistent (across different models) step.
```

Comment:
Slight rephrasing in `...more complex models required more complex preprocessing` to `some models required more complex preprocessing`.